internal/linuxint/types.go

```diff
@@ -0,0 +1,107 @@
+// Package linux provides type information for the current kernel.
+package linuxint
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/cilium/ebpf/btf"
+	"github.com/cilium/ebpf/internal"
+)
+
+var kernelBTF struct {
```

Comment:
To our discussion of earlier: in the future, instead of the anonymous struct `kernelBTF`, we could have a hypothetical `btf.Loader` here that acts as a cache and serves as the API to load vmlinux and kmods. I see this package already imports `btf/`, so no need for a global `any` var.

The `btf.Loader` stored in this package would become the internal copy, it would be flushed by `linux.FlushTypes()`, and `linux.Types()` would instead return a `btf.Loader` so the caller can have its own personal BTF cache.

With this PR, it doesn't seem like package `btf/` still relies on accessing the kernel spec. That makes things a lot less complicated!

Reply:
> a hypothetical btf.Loader here that acts as a cache and serves as the API to load vmlinux
 
Yup! This would parse BTF more often than what we currently have though: once when the library first needs vmlinux and once for any other `btf.Loader`. The loader would make it easier to share a single copy, but would still lead to multiple copies in a single binary. Maybe that is fine though for users that care about memory pressure?

> and kmods.

This is much trickier, since kmods can come and go! Not sure yet how to deal with this.